services/twitch/twitch.service.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const serverSecrets = require('../../lib/server-secrets')
+const { BaseJsonService, NotFound } = require('..')
+
+const schema = Joi.any()
+
+class TwitchStatus extends BaseJsonService {
+  static get category() {
+    return 'activity'
+  }
+
+  static get route() {
+    return {
+      base: 'twitch/status',
+      pattern: ':user',
+    }
+  }
+
+  static get examples() {
+    const preview = this.render({
+      user: 'andyonthewings',
+      isLive: true,
+    })
+    delete preview.link
+    return [
+      {
+        title: 'Twitch Status',
+        namedParams: {
+          user: 'andyonthewings',
+        },
+        // hard code the static preview
+        // because link[] is not allowed in examples
+        staticPreview: preview,
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'twitch',
+    }
+  }
+
+  static render({ user, isLive }) {
+    return {
+      message: isLive ? 'live' : 'offline',
+      color: isLive ? 'red' : 'lightgrey',
+      link: `https://www.twitch.tv/${user}`,
+    }
+  }
+
+  async fetch({ user }) {
+    const headers = { 'Client-ID': serverSecrets.twitch_client_id }
+    const users = await this._requestJson({
+      schema,
+      url: `https://api.twitch.tv/helix/users`,
+      options: {
+        headers,
+        qs: { login: user },
+      },
+    })
+    if (users.data.length < 1) {
+      throw new NotFound({ prettyMessage: 'invalid user' })
+    }
```

Comment:
Is it necessary to make these two API calls? What kind of response would we get from the https://api.twitch.tv/helix/streams API if `user` is invalid?

Reply:
If `user` does not exist on Twitch, the result would be an empty array, which is the same as when a user is offline.

Reply:
Ah okay, that makes sense. Maybe it would be good to add an inline comment so that future readers of the code have that context?